### PR TITLE
Lint and build improvements

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

* Adds support for class properties (Babel + ESLint)
* Adds support for private class properties (Babel + ESLint)
* Removes unused dependencies
* Set the proper stats level for Webpack (Prevent the diarhea of info in the console when building the client)

**Issues**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,13 +6,15 @@
...
+    "@babel/plugin-proposal-class-properties": "^7.10.1",
+    "@babel/plugin-proposal-private-methods": "^7.10.1",
     "@babel/plugin-transform-modules-amd": "^7.9.6",
     "@babel/polyfill": "^7.8.7",
     "@babel/preset-env": "^7.8.6",
     "autoprefixer": "^9.8.0",
+    "babel-eslint": "^11.0.0-beta.2",
     "babel-loader": "^8.0.6",
     "browser-sync": "^2.26.7",
-    "clean-webpack-plugin": "^3.0.0",
```
Comment (cromefire) [inline package.json:15]: Is there a reason for that? I'd consider it good to cleanup the results before the build (so there are no remaining files that don't belong there)

File: webpack.common.js [hunk 2]
```diff
@@ -16,6 +16,7 @@ const Assets = [
 module.exports = {
     context: path.resolve(__dirname, 'src'),
     entry: './bundle.js',
+    stats: 'errors-only',
```
Comment (cromefire) [inline webpack.common.js:19]: What's up with this wouldn't it be better to set that by cli if needed

File: package.json [hunk 3]
```diff
@@ -47,9 +49,6 @@
     "stylelint-no-browser-hacks": "^1.2.1",
     "stylelint-order": "^4.0.0",
     "webpack": "^4.41.5",
-    "webpack-cli": "^3.3.10",
```
Comment (cromefire) [inline package.json:50]: This should be needed right?

Comment (cromefire) [reply package.json:50]: Also it seems to be an old version

File: package.json (hunk 1 above)
Comment (MrTimscampi) [reply package.json:15]: That's handled by Gulp.

File: package.json (hunk 3 above)
Comment (cromefire) [reply package.json:50]: https://webpack.js.org/guides/getting-started/#basic-setup

File: webpack.common.js (hunk 2 above)
Comment (MrTimscampi) [reply webpack.common.js:19]: It's meant to hide all the non essential output of webpack (stats about bundle size and so on) during builds

File: package.json (hunk 3 above)
Comment (MrTimscampi) [reply package.json:50]: It's not used, webpack is called through Gulp.

**Pull request merged**